dev-docs/source/Gatekeeping.rst

```diff
@@ -0,0 +1,156 @@
+.. _Gatekeeping:
+
+==========================
+Gatekeeping a Pull Request
+==========================
+
+The final step in our :ref:`development workflow <GitWorkflow>` after a :ref:`pull request has been reviewed <ReviewingAPullRequest>` is gatekeeping. This is second review stage where a senior developer / code owner does a final check of the work before merging it to the ``main`` branch.
```

Comment:
Are there any additional things we need to pay attention to before merging a PR into the `release-next` branch?

Reply:
We are very careful these days that no new features get merged into `release-next`. It really should only be bugfixes that have been identified as regressions for that release. Anything else should be merged into main for the following release (although it will be available in the nightly after release when we switch back to nightly being published from the main branch). 

I think being aware of where we are in the release sprint is probably a good thing. We reach a point where we are ready to release but are still building the packages and testing so `release-next` exists but is no longer being auto merged into main. As such any merges into `release-next` after this point would be lost entirely.

Reply:
It probably would be good to have a section in this document explaining that.

Reply:
Good point, I'll add one